cpp/arcticdb/version/version_map_entry.hpp

```diff
@@ -83,6 +80,63 @@ struct LoadParameter {
     }
 };
 
+
+inline bool is_undeleted_strategy_subset(const LoadStrategy& left, const LoadStrategy& right){
+    switch (left.load_type_) {
+        case LoadType::NOT_LOADED:
+            return true;
+        case LoadType::LOAD_LATEST:
+            // LOAD_LATEST is not a subset of LOAD_DOWNTO because LOAD_DOWNTO may not reach the latest undeleted version.
+            return right.load_type_ != LoadType::NOT_LOADED && right.load_type_ != LoadType::LOAD_DOWNTO;
+        case LoadType::LOAD_DOWNTO:
+            if (right.load_type_ == LoadType::LOAD_ALL) {
+                return true;
+            }
+            if (right.load_type_ == LoadType::LOAD_DOWNTO && ((left.load_until_version_.value()>=0) == (right.load_until_version_.value()>=0))) {
+                // Left is subset of right only when the [load_until]s have same sign and left's version is >= right's version
```

Comment:
I don't think this is right when they're both negative

Reply:
I think it should be correct. E.g. `-2 >= -3` and `-2` will load only 2 versions whereas the `-3` will load 3 versions which means `-2` will be a subset of `-3`.

Reply:
I think this implementation is correct since we want these semantics:

left v2 right v3 => left not subset of right

left v3 right v2 => left subset of right

left v-1 right v-2 => left subset of right (left is HEAD, right is HEAD~)

left v-2 right v-1  => left not subset of right

Reply:
Agreed, ignore me